lighthouse-core/gather/driver.js

```diff
@@ -1581,6 +1586,150 @@ class Driver {
+    this._onRequestPausedHandlers.set(url, handler);
+  }
+
+  /**
+   * @param {LH.Crdp.Fetch.RequestPausedEvent} event
+   */
+  async _onRequestPaused(event) {
+    const handler = this._onRequestPausedHandlers.get(event.request.url);
+    if (handler) {
+      await handler(event);
+    } else {
+      // Nothing cares about this URL, so continue.
+      await this.sendCommand('Fetch.continueRequest', {requestId: event.requestId});
+    }
+  }
+
+  async disableRequestInterception() {
+    await this.sendCommand('Fetch.disable');
+    await this.off('Fetch.requestPaused', this._onRequestPaused);
+    this._onRequestPausedHandlers.clear();
+  }
+
+  /**
+   * Requires that `driver.enableRequestInterception` has been called.
+   *
+   * Fetches any resource in a way that circumvents CORS.
+   *
+   * @param {string} url
+   * @param {number} timeoutInMs
+   * @return {Promise<string>}
+   */
+  async fetchArbitraryResource(url, timeoutInMs = 500) {
+    if (!this.isDomainEnabled('Fetch')) {
+      throw new Error('Fetch domain must be enabled to use fetchArbitraryResource');
+    }
+
+    /** @type {Promise<string>} */
+    const requestInterceptionPromise = new Promise((resolve, reject) => {
+      this.setOnRequestPausedHandler(url, async (event) => {
+        const {requestId, responseStatusCode} = event;
+
+        // The first requestPaused event is for the request stage. Continue it.
+        if (!responseStatusCode) {
+          // Remove same-site cookies so we aren't buying stuff on Amazon.
+          const sameSiteCookies = await this.sendCommand('Network.getCookies', {urls: [url]});
+          const sameSiteCookiesKeyValueSet = new Set();
+          for (const cookie of sameSiteCookies.cookies) {
+            sameSiteCookiesKeyValueSet.add(cookie.name + '=' + cookie.value);
+          }
+          const strippedCookies = event.request.headers['Cookie']
+            .split(';')
+            .filter(cookieKeyValue => {
+              return !sameSiteCookiesKeyValueSet.has(cookieKeyValue.trim());
+            })
+            .join('; ');
+
+          this.sendCommand('Fetch.continueRequest', {
+            requestId,
+            headers: [{name: 'Cookie', value: strippedCookies}],
+          });
+          return;
+        }
+
+        // Now in the response stage, but the request failed.
+        if (!(responseStatusCode >= 200 && responseStatusCode < 300)) {
+          reject(new Error(`Invalid response status code: ${responseStatusCode}`));
+          return;
+        }
+
+        const responseBody = await this.sendCommand('Fetch.getResponseBody', {requestId});
+        if (responseBody.base64Encoded) {
+          resolve(Buffer.from(responseBody.body, 'base64').toString());
+        } else {
+          resolve(responseBody.body);
+        }
+
+        // Fail the request (from the page's perspective) so that the iframe never loads.
+        this.sendCommand('Fetch.failRequest', {requestId, errorReason: 'Aborted'});
```

Comment:
`Aborted by Lighthouse`?

Reply:
this is an enum, can't give custom messages.